### PR TITLE
Update navbar

### PR DESCRIPTION
**Pull request by johrstrom**
https://osc.github.io/ood-documentation-test/update-navbar/

A lot of updates around modifying the navbar. some of it's moving sections around to fit a uniform description other updates are more around the actual content.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1135148780858012/1203979985450112) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by johrstrom** [comment]
@abujeda FYI, I'm making some major updates/refactoring to this entire section on controlling the navbar.  Still a work in progress, that's why it's in draft.  I'll let you know when it's ready and ask if you could give feedback on it.

File: source/customizations/main-navigation.inc [hunk 1]
```diff
@@ -1,205 +1,372 @@
...
-
-  When configured as a ``links`` array item, the subcategory will not be used to group the matched applications.
-  The definition order is respected when rendering the links.
-  If grouping is required, use the ``group`` property to add the desired structure as shown in the
-  :ref:`applications based menu above. <applications_based_menu>`
+        - "sys/Rstudio/*"
+        - "sys/bc_desktop"
+
+
+**IMAGE NEEDED**
```
Comment (johrstrom) [inline source/customizations/main-navigation.inc]: I need an image here for this config

```yaml

  nav_bar:
    - title: "My Apps"
      icon: "fa://code"
      links:
      links:
        - "sys/Rstudio/*"
        - "sys/bc_desktop"
```

**Comment by johrstrom** [comment]
@abujeda this is ready for you to look over. I took a lot of liberty to restructure it a bit, here are some highlights

* first, the order of these entries are simplest to most complex
* The first 2 under `fully customizing the navbar` center around menus for applications. The first on using `apps` the second using `links` and reworded the notes on how using `links` differs from using `apps` while combining the sections on using the optional `groups`.
* I combined some things into 1 entry entitled `Menus based on links` and added links specifically for profiles and urls given other pages link to them

**Comment by johrstrom** [comment]
BTW you should be able to see the resulting html at https://osc.github.io/ood-documentation-test/update-navbar/ in case you don't want to build it yourself.

File: source/customizations/main-navigation.inc (hunk 1 above)
Comment (abujeda) [reply source/customizations/main-navigation.inc]: Sure - Here it is:

def-menu-links

File: source/customizations.rst [hunk 2]
```diff
@@ -187,7 +187,7 @@ Since OnDemand 2.1, custom links can be added to the Help menu using the configu
 Links will be inserted at the end of the core links already included in the menu by the OnDemand codebase.
 
 ``help_menu`` supports all the link definitions developed for the custom navigation configuration.
-For more information on how to create custom links, see the :ref:`link definition documentation <navbar_links_guide>`.
+For more information on how to create custom links, see for example `adding urls to menus <menus-for-urls>`.
```
Comment (abujeda) [inline source/customizations.rst]: Should this be a :`ref:` to generate a link to the `menus-for-urls `section?
Currently is rendering as regular text.

File: source/customizations/profiles.inc [hunk 3]
```diff
@@ -52,7 +52,8 @@ Profile selection can be done using configurable navigation links.
 These links will update the profile, save the newly selected profile in the user settings file and reload the Dashboard. 
 
 To add profile links to the Help navigation dropdown menu use the configuration property ``profile_links``.
-For more information on how to configure profile links, see the :ref:`link definition documentation. <navbar_links_guide>`.
+For more information on how to add profile links to the navigation bar in various places, see
+`menus for profile links <menus-for-profile links>`.
```
Comment (abujeda) [inline source/customizations/profiles.inc]: Same as above: should this be a `:ref:` to generate a link?
Currently is rendering as regular text.

File: source/release-notes/v1.5-release-notes.rst [hunk 4]
```diff
@@ -166,7 +166,7 @@ Upgrading from v1.3
    As a result menus may appear that you do not expect, if previously you were relying
    on the "whitelist" functionality of the ``NavConfig.categories``.
 
-   See :ref:`dashboard-navbar-config` for details.
+   See  `Control which apps appear in the Dashboard Navbar <1-4-dashboard-navbar-config>`_ for details.
```
Comment (abujeda) [inline source/release-notes/v1.5-release-notes.rst]: Same as above

File: source/release-notes/v1.4-release-notes.rst [hunk 5]
```diff
@@ -227,7 +227,7 @@ Improve default discoverability of apps in the Dashboard's navbar
 
 Introduction of a whitelist mode for the Dashboard navbar which is disabled by default. This change means that by default, when deploying a new app, if properly configured it will appear in the Dashboard's navigation menu without the need for changing configuration. `OSC/ood-dashboard#295 <https://github.com/OSC/ood-dashboard/issues/295>`__
 
-For details please see :ref:`dashboard-navbar-config`.
+For details please see `Control which apps appear in the Dashboard Navbar <1-4-dashboard-navbar-config>`_.
```
Comment (abujeda) [inline source/release-notes/v1.4-release-notes.rst]: Same as above

**Comment by abujeda** [comment]
LGTM. Apart from the references not rendering as links, all else looks good.

**Comment by johrstrom** [comment]
Thanks for looking at it for me!

**Pull request merged**